server.go

```diff
@@ -0,0 +1,75 @@
+package unixtransport
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+)
+
+// A TestServer is an HTTP+UNIX server listening on a temporary UNIX socket, for
+// use in end-to-end HTTP tests. TestServer's API mimics that of the
+// net/http/httptest.Server type.
+type TestServer struct {
+	// Base URL of form http+unix:///tmp/(random)/test.sock: with no trailing
+	// slash.
+	URL string
+
+	server *httptest.Server
+	client *http.Client
+
+	// Temporary directory, cleaned up on close.
+	dir string
+}
+
+// NewTestServer starts and returns a new TestServer. The caller should call
+// Close when finished, to shut it down.
+func NewTestServer(handler http.Handler) *TestServer {
+	// If any subsequent calls fail before we return *TestServer, be sure to
+	// os.RemoveAll(dir) to tidy things up.
+	dir, err := os.MkdirTemp(os.TempDir(), "unixtransport-server")
+	if err != nil {
+		panicf("failed to make temporary directory: %v", err)
+	}
+
+	l, err := net.Listen("unix", filepath.Join(dir, "test.sock"))
+	if err != nil {
+		_ = os.RemoveAll(dir)
+		panicf("failed to listen on UNIX socket: %v", err)
+	}
+
+	// Offload the heavy-lifting to httptest.Server and its methods.
+	server := httptest.NewUnstartedServer(handler)
+	server.Listener = l
+	server.Start()
+
+	// And finally create an http.Client which can speak http+unix scheme.
+	client := server.Client()
+	Register(client.Transport.(*http.Transport))
+
+	return &TestServer{
+		URL: fmt.Sprintf("http+unix://%s:", l.Addr().String()),
```

Comment:
Is this correct? The colon seems suspect?

Reply:
Yep, it is. I am using the format you documented for compatibility with the modified transport:
```
=== RUN   TestClientPush/UNIX
(string) (len=57) "http+unix:///tmp/unixtransport-server815910969/test.sock:"
(string) (len=62) "http+unix:///tmp/unixtransport-server815910969/test.sock:/push"
```

The first is raw `srv.URL`, the second is with my specific handler path.

Reply:
Though I guess the question is: should I be required to add the `:/push` suffix in my own code?

Reply:
Ah, I see what you're doing here. I think this is a reasonable approach. The more formal method would be for the package to offer some URL-building functionality, but I'm not sure that juice is worth the squeeze.